packages/nextjs/services/database/scripts/update-ens.ts

```diff
@@ -0,0 +1,213 @@
+        fallbackAddresses.push(address);
+      }
+    });
+
+    // Handle fallbacks using viem's getEnsName
+    if (fallbackAddresses.length > 0) {
+      console.log(`Falling back to getEnsName for ${fallbackAddresses.length} addresses with failed resolvers`);
+
+      // Process fallbacks in smaller batches to avoid rate limits
+      const fallbackChunks = chunk<`0x${string}`>(fallbackAddresses, 10);
+      for (const fallbackBatch of fallbackChunks) {
+        await Promise.all(
+          fallbackBatch.map(async (address: `0x${string}`) => {
+            try {
+              const ensName = await publicClient.getEnsName({ address });
+              if (ensName) {
+                results[address] = ensName;
+                console.log(`Found ENS name via fallback for ${address}: ${ensName}`);
+              } else {
+                results[address] = null;
+              }
+            } catch (error) {
+              console.error(`Error in fallback lookup for ${address}:`, error);
+              results[address] = null;
+            }
+          }),
+        );
+        // Delay between fallback chunks
+        await new Promise(resolve => setTimeout(resolve, 100));
+      }
+    }
+
+    // Delay between main batches to avoid rate limiting
+    await new Promise(resolve => setTimeout(resolve, 200));
+  }
+
+  // Log summary
+  console.log("\nENS Lookup Summary:");
+  console.log(`Total addresses processed: ${addresses.length}`);
+  console.log(`Addresses with no ENS records: ${noEnsRecords.length}`);
+  console.log(`Addresses with failed resolvers: ${failedResolvers.length}`);
+  console.log(`Successfully resolved ENS names: ${Object.values(results).filter(Boolean).length}`);
+
+  return results;
+}
+
+async function updateEnsNames() {
+  try {
+    const usersWithoutEns = await db.query.users.findMany({
+      where: isNull(users.ens),
+    });
+
+    console.log(`Found ${usersWithoutEns.length} users without ENS names`);
+
+    // Get all addresses that need ENS lookup
+    const addresses = usersWithoutEns.map(user => user.userAddress as `0x${string}`);
+
+    // Batch process ENS lookups
+    const ensResults = await batchGetENS(addresses);
+
+    // Prepare bulk update data
+    const updates = Object.entries(ensResults)
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
+      .filter(([_, ensName]) => ensName !== null)
+      .map(([address, ensName]) => ({
+        address: address.toLowerCase(),
+        ens: ensName,
+      }));
+
+    if (updates.length > 0) {
+      // Perform bulk update
+      await db.transaction(async tx => {
+        for (const { address, ens } of updates) {
+          await tx
+            .update(users)
+            .set({
+              ens,
+              updatedAt: new Date(),
+            })
+            .where(eq(lower(users.userAddress), address));
+        }
+      });
+      console.log(`Bulk updated ENS names for ${updates.length} users`);
+    }
```

Comment:
Transaction is not supported for neon-http. Getting an error when trying to execute this against staging database (Also, transactions just run requests one at a time)

You can check the `0001_user_data.ts` from bg-imports. I tried this and it works:

```
      const updateQueries = updates.map(({ address, ens }) =>
        db
          .update(users)
          .set({
            ens,
            updatedAt: new Date(),
          })
          .where(eq(lower(users.userAddress), address)),
      );

      await db.executeQueries(updateQueries as any);
```

We created `executeQueries` in `postgresClient`  so we could run the ALL queries in a single call.